stripe-core/src/main/java/com/stripe/android/core/exception/ExceptionUtils.kt

```diff
@@ -0,0 +1,12 @@
+@file:RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+
+package com.stripe.android.core.exception
+
+import androidx.annotation.RestrictTo
+
+@get:RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+val Throwable.safeAnalyticsMessage: String
+    get() = when (this) {
+        is StripeException -> analyticsValue()
+        else -> this::class.java.simpleName
```

Comment:
Will this cause noise than the help it'll provide? Classes will likely end up minified/obfuscated.

I think this makes sense for java exceptions like IOException, etc. But if we're expecting other exceptions, this will cause a lot of noise.

Reply:
We shouldn't be expecting anything much else besides `IOException` and the `StripeException` subclasses. I could eliminate that noise completely by checking if it can be logged.

```kotlin
@get:RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
val Throwable.canLog: Boolean
    get() = this is StripeException

@get:RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
val Throwable.safeAnalyticsMessage: String
    get() = when (this) {
        is StripeException -> analyticsValue()
        else -> "unknown"
    }
```